File: core/src/software/aws/toolkits/core/telemetry/TelemetryPublisher.kt
// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package software.aws.toolkits.core.telemetry

import software.amazon.awssdk.services.toolkittelemetry.model.Sentiment

interface TelemetryPublisher {
    suspend fun publish(metricEvents: Collection<MetricEvent>)

    suspend fun sendFeedback(sentiment: Sentiment, comment: String)
}

File: src/shared/telemetry/telemetryPublisher.ts
/*!
 * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

import { MetricDatum } from './clienttelemetry'
import { TelemetryFeedback } from './telemetryFeedback'

export interface TelemetryPublisher {
    init(): Promise<void>

    postFeedback(feedback: TelemetryFeedback): Promise<void>

    enqueue(...events: MetricDatum[]): any
    flush(): Promise<any>
}
